### DISCUSSION TITLE
On clicking devices listed on rack, device URL redirect happens with http instead of https

### DISCUSSION THREAD

**Discussion by NagaKarthi** [netbox-community/netbox#11523] [General]
Hi,

- Python Version: 3.9.10
- Netbox Version: v3.3.7

On the Racks, If I click on any saved device in the rack(Front or Rear), the URL redirect to the device page is happening with http instead of https. This behavior is found on existing device and also in addition of a new device from rack page itself. So far I was able to find this problem only with redirect to device from rack page.

Current Setup I have is:
ELB <----HTTPS to HTTP-----> EC2(Netbox)

**The Below is my nginx config:**

server {
    listen 80;

    # CHANGE THIS TO YOUR SERVER'S NAME
    server_name XX.XX.XX.XX(My ip);

    ssl_certificate /etc/ssl/certs/netbox.crt;
    ssl_certificate_key /etc/ssl/private/netbox.key;

    client_max_body_size 25m;

    location /static/ {
        alias /opt/netbox/netbox/static/;
    }

    location / {
        proxy_pass http://127.0.0.1:8001;
        proxy_set_header X-Forwarded-Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}

server {
    # Redirect HTTP traffic to HTTPS
    listen 80;
    server_name _;
    return 301 https://$host$request_uri;
}

**Comment by markkuleinio** [comment]
Since you are using ELB to do TLS for you (and it uses plain http to NetBox EC2), you need to set `X-Forwarded-Proto` to `https` in your nginx config.

```
    proxy_set_header X-Forwarded-Proto https;
```

Note that there is no use configuring the `ssl_certificate*` options in nginx when there is no TLS in nginx at all in this case.

**Reply by NagaKarthi** [reply]
Thanks @markkuleinio, the above solution fixed the redirect issues. Also as suggested I've removed the `ssl_certificate*` statement from nginx config.
